fastlane_bot/helpers/carbon_trade_splitter.py

```diff
@@ -1,203 +1,71 @@
-            token_type: the self.NATIVE, self.WRAPPED, or SELF.NEITHER
-
-        Returns:
-            The correct token address for the pool.
-
-        """
-        if token_address in [
-            self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS,
-            self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS,
-        ]:
-            return self.ConfigObj.NATIVE_GAS_TOKEN_ADDRESS if token_type == self.NATIVE else self.ConfigObj.WRAPPED_GAS_TOKEN_ADDRESS
-        return token_address
-
-    def _extract_carbon_trades(self, trade) -> List[TradeInstruction]:
-        """
-        This function calls the functions containing the logic to extract Carbon trades.
-
-        Args:
-            trade: A single TradeInstruction object.
-
-        Returns:
-            A list of TradeInstruction objects.
+    Args:
+        - `cfg`: the configuration object.
+        - `trade_instructions`: A list of trade instructions.
 
-        """
-        carbon_exchanges = self._process_carbon_trades(trade)
-        return self._create_new_trades_from_carbon_exchanges(
-            carbon_exchanges, trade
-        )
+    Returns:
+        - `new_trade_instructions`: A new list of trade instructions.
+    """
 
-    def _process_carbon_trades(self, trade: TradeInstruction) -> Dict:
-        """Separates Carbon trades by exchange & wrapped/native token address.
+    new_trade_instructions = []
+    for trade_instruction in trade_instructions:
+        if trade_instruction.exchange_name not in cfg.CARBON_V1_FORKS:
+            new_trade_instructions.append(trade_instruction)
+            continue
 
-        Processes Carbon trades and organizes them by exchange and token type.
-
-        Args:
-            trade: a single TradeInstruction object
-
-        Returns:
-            A dictionary containing a list of
-
-        """
         carbon_exchanges = {}
 
-        raw_tx_str = trade.raw_txs.replace("'", '"').replace('Decimal("', '').replace('")', '')
-        raw_txs = json.loads(raw_tx_str)
-
-        for _tx in raw_txs:
-            curve = trade.db.get_pool(cid=str(_tx["cid"]).split("-")[0])
-            exchange = curve.exchange_name
-            token_type = self._get_token_type(curve)
-
-            if exchange not in carbon_exchanges:
-                carbon_exchanges[exchange] = self._initialize_exchange_data()
-
-            self._update_exchange_data(
-                carbon_exchanges[exchange], token_type, _tx, trade
+        for tx in loads(trade_instruction.raw_txs.replace("'", '"').replace('Decimal("', '').replace('")', '')):
+            pool = trade_instruction.db.get_pool(cid=str(tx["cid"]).split("-")[0])
+
+            if cfg.NATIVE_GAS_TOKEN_ADDRESS in pool.get_tokens:
+                token_type = cfg.NATIVE_GAS_TOKEN_ADDRESS
+            elif cfg.WRAPPED_GAS_TOKEN_ADDRESS in pool.get_tokens:
+                token_type = cfg.WRAPPED_GAS_TOKEN_ADDRESS
+            else:
+                token_type = cfg.ZERO_ADDRESS
```

Comment:
Yeah, it's z kind of an abuse I suppose. It can be anything other than those two other addresses (which are not even used as addresses, but rather as unique identifiers; perhaps that's the real abuse).
In either case, it may well be just an empty string, since the zero address is indeed kinda meaningless here (in the sense of what the zero address is typically used for).
I'll change it to `''`, since it is referred to as a string elsewhere in the code, and keeping a strict-typing convention is of the essence.
Tx.